src/pymor/algorithms/system.py

```diff
@@ -0,0 +1,227 @@
+
+    @match_class(BlockRowOperator)
+    def action_BlockRowOperator(self, op):
+        range_bases, source_bases, system_in_range, system_in_source, products = \
+            self.range_bases, self.source_bases, self.system_in_range, self.system_in_source, self.products
+        assert not system_in_range
+        assert system_in_source  # Other case is handled by action_no_system
+
+        def project_block(j, block):
+            return project(block, range_bases, source_bases[j], products)
+
+        projected_blocks = [project_block(j, o) if not isinstance(o, ZeroOperator) else None
+                            for j, o in enumerate(op.blocks)]
+        return BlockRowOperator(projected_blocks)
+
+    @match_class(BlockColumnOperator)
+    def action_BlockColumnOperator(self, op):
+        range_bases, source_bases, system_in_range, system_in_source, products = \
+            self.range_bases, self.source_bases, self.system_in_range, self.system_in_source, self.products
+        assert not system_in_source
+        assert system_in_range  # Other case is handled by action_no_system
+
+        def project_block(i, block):
+            return project(block, range_bases[i], source_bases, product=products[i])
+
+        projected_blocks = [project_block(i, o) if not isinstance(o, ZeroOperator) else None
+                            for i, o in enumerate(op.blocks)]
+        return BlockColumnOperator(projected_blocks)
+
+    @match_class(VectorArrayOperator)
+    def action_VectorArrayOperator(self, op):
+        range_bases, source_bases, system_in_range, system_in_source, products = \
+            self.range_bases, self.source_bases, self.system_in_range, self.system_in_source, self.products
+        if not op.transposed:
+            assert not system_in_source
+            if source_bases is not None:
+                raise NotImplementedError
+            assert system_in_range  # Other case is handled by action_no_system
+
+            def inner(U, V, product=None):
+                return U.dot(V) if product is None else product.apply2(U, V)
+
+            projected_array = BlockVectorSpace.make_array(
+                [NumpyVectorSpace.from_data(inner(block, range_basis, product), block.space.id)
+                 for block, range_basis, product in zip(op._array._blocks, range_bases, products)]
+            )
+        else:
+            assert not system_in_range
+            if range_bases is not None:
+                raise NotImplementedError
+            assert system_in_source  # Other case is handled by action_no_system
+
+            projected_array = BlockVectorSpace.make_array(
+                [NumpyVectorSpace.from_data(block.dot(source_basis), block.space.id)
+                 for block, source_basis in zip(op._array._blocks, source_bases)]
+            )
+        return VectorArrayOperator(projected_array, transposed=op.transposed, space_id=op.space_id,
+                                   name=op.name)
```

Comment:
for newer pymor versions `transposed` is replaced by `adjoint`

Reply:
Right, thanks!